### ISSUE TITLE
[5.7.2] Crash when content-disposition header has wrong format

### ISSUE THREAD

**Issue by jloewe** [Ultimaker/Cura#19371]
### Cura Version

5.7.2

### Operating System

macOS 14.5

### Printer

Ender 3

### Reproduction steps

Open `cura://open?file=https://files.printables.com/media/prints/781568/stls/6068683_fb60a840-f676-4524-8e5f-568b80a448eb_f0da1d87-6615-498b-98a4-0045831df6cd/ikea-samla-45l-spool-holder-frame-tl-br.stl`

### Actual results

Cura crashes with the following error.

```
Traceback (most recent call last):
  File "UM/TaskManagement/TaskManager.py", line 142, in event
    call_event.callFunction()
  File "UM/TaskManagement/TaskManager.py", line 33, in callFunction
    self._function(*self._args, **self._kwargs)
  File "cura/CuraApplication.py", line 1911, in on_finish
    assert content_disposition_match is not None
AssertionError
```

### Expected results

The stl is loaded into Cura.

### Add your .zip and screenshots here ⬇️

The `on_finish` function inside of the `_openURL` function is causing this crash.

https://github.com/Ultimaker/Cura/blob/935109680357710057f8bc1b6cf2dec3e9d317da/cura/CuraApplication.py#L1895-L1920

The crash occurs because the printables content-disposition header looks like this:

```
attachment; filename*=utf-8''Ikea%20Samla%2045L%20Spool%20Holder%20-%20Frame%20TL-BR.stl
```

Cura expects the header to be in the following format:

```regex
attachment; filename="(?P<filename>.*)"
```

A possible solution would be to rearrange the checks and remove the assertion that causes the crash when there is no match in the header. The fallback used when there is no header could also be used when there is no match.

**Comment by jloewe** [comment]
This was fixed in 38b496102ff449d9f6ff513e7d8a5bc248693357.

**Comment by jloewe** [comment]
This was fixed in 38b496102ff449d9f6ff513e7d8a5bc248693357.
